pkg/reader/ccip.go

```diff
@@ -1292,5 +1294,71 @@ func (r *ccipChainReader) getRMNRemoteAddress(
 	return rmnRemoteAddress, nil
 }
 
+// Get the DestChainConfig from the FeeQuoter contract on the given chain.
+func (r *ccipChainReader) getFeeQuoterDestChainConfig(
+	ctx context.Context,
+	chainSelector cciptypes.ChainSelector,
+) (cciptypes.FeeQuoterDestChainConfig, error) {
+	if err := validateExtendedReaderExistence(r.contractReaders, chainSelector); err != nil {
+		return cciptypes.FeeQuoterDestChainConfig{}, err
+	}
+
+	var destChainConfig cciptypes.FeeQuoterDestChainConfig
+	srcReader := r.contractReaders[chainSelector]
+	err := srcReader.ExtendedGetLatestValue(
+		ctx,
+		consts.ContractNameFeeQuoter,
+		consts.MethodNameGetDestChainConfig,
+		primitives.Unconfirmed,
+		map[string]any{
+			"destChainSelector": r.destChain,
+		},
+		&destChainConfig,
+	)
+
+	if err != nil {
+		return cciptypes.FeeQuoterDestChainConfig{}, fmt.Errorf("failed to get dest chain config: %w", err)
+	}
+
+	return destChainConfig, nil
+}
+
+// GetMedianDataAvailabilityGasConfig returns the median of the DataAvailabilityGasConfig values from all FeeQuoters
+func (r *ccipChainReader) GetMedianDataAvailabilityGasConfig(
```

Comment:
Some doc clarifying the decision behind it would be great. Why do we get all the configs? Why do compute a median from them etc